### ISSUE TITLE
Salt state.single returns retcode 0 when state function does not exist

### ISSUE THREAD

**Issue by nasenbaer13** [saltstack/salt#33672]
### Description of Issue

The event returned from a state.single module execution has `retcode` set to 0 despite the fact, that the function argument does not map to a valid function. This in particular affects orchestration runs, which cannot detect the error for this case.

```
state_not_exists:
  salt.function:
    - name: state.single
    - tgt: testminion
    - arg:
      - state.doesnotexist
      - state_not_exists

```

### Setup

latest develop branch

### Steps to Reproduce Issue

To reproduce the problem , one does not need an orchestration state. Looking at the event bus is sufficient.

```
salt testminion state.single muh.maeh somename
testminion:
    Data failed to compile:
----------
    Specified state 'muh.maeh' was not found
```

at the same time / monitor the salt event bus
```
salt-run state.event
salt/job/20160601164315705689/ret/testminion	{"fun_args": ["muh.maeh", "somename"], "jid": "20160601164315705689", "return": ["Specified state 'muh.maeh' was not found"], "retcode": 0, "success": true, "cmd": "_return", "_stamp": "2016-06-01T14:43:23.602401", "fun": "state.single", "id": "testminion", "out": "highstate"}
```
### Versions Report

```
salt --versions-report
Salt Version:
           Salt: 2016.3.0-629-g941bd1d
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.11 (default, Jan 21 2016, 18:32:00)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.2
 
System Versions:
           dist:   
        machine: x86_64
        release: 15.5.0
         system: Darwin
        version: 10.11.5 x86_64
```

**Comment by Ch3LL** [comment]
@nasenbaer13 I am also seeing this behavior on 2015.5.9 2015.8.10 and 2016.3.0. 

```
    "id": "ch3ll-master", 
    "jid": "20160601155027867173", 
    "out": "highstate", 
    "retcode": 0, 
    "return": [
        "Specified state 'muh.maeh' was not found"
    ], 
```

Looks like we need to fix this up to not return 0 if there is an error. Thanks!

**Comment by rallytime** [comment]
@nasenbaer13 I think we have some good news here. A fix for this was just pushed into develop in #33695. My testing shows the following:

Pre-fix on develop:
```
root@rallytime:~# salt rallytime state.single muh.maeh not_exist
rallytime:
    Data failed to compile:
----------
    Specified state 'muh.maeh' was not found
root@rallytime:~# echo $?
0
```
Post-fix on develop:
```
root@rallytime:~# salt rallytime state.single muh.maeh not_exist
rallytime:
    Data failed to compile:
----------
    Specified state 'muh.maeh' was not found
ERROR: Minions returned with non-zero exit code
root@rallytime:~# echo $?
11
```
Can you confirm that the HEAD of the current develop fixes the `state.single` issue for you?

**Comment by nasenbaer13** [comment]
@rallytime ack - I get the same result. Also the orchestrator now fails with the correct message. Can the fix be backported to 2015.8?

**Comment by rallytime** [comment]
@nasenbaer13 I am hesitant to backport that fix to 2015.8 as it touches important places of the code and it'd probably be better to let that sit and bake in develop for a while. Unless @cachedout has a different opinion?

**Comment by nasenbaer13** [comment]
OK, no problem. As long as it is not merged to 2016.3.x t probably makes no sense to pull it in 2015.8.x - so I'll close the issue.
